### DISCUSSION TITLE
Add earthaccess as a dependency for CMR search functionality?

### DISCUSSION THREAD

**Discussion by scottyhq** [SlideRuleEarth/sliderule#431] [General]
Attempting to consolidate some discussion from UW Hackweek, Slack, etc. There is a general interest in ensuring various Python tools are working well together and not duplicating code and effort! One opportunity for this is to consolidate CMR search functionality:

* https://github.com/nsidc/earthaccess is emerging as the leading client library for authentication, search, and optionally download/stream data from CMR. As a dependency it in turn uses [python-cmr](https://github.com/nasa/python_cmr), although there is discussion here about the merits of alternatively using CMR-STAC https://github.com/nsidc/earthaccess/discussions/221. 

* sliderule-python has kept around some client-side code for querying CMR https://slideruleearth.io/rtd/api_reference/earthdata.html and separately uses server-side code to interact with CMR. Instead of custom code, `earthaccess` could be used for search, and possibly there are some advanced optimizations that could be upstreamed to earthaccess. 

* There is a bit of a challenge right now in getting *footprint polygons* right now from earthaccess. If using CMR-STAC this is easy but maybe slow compared to custom parsing polygons from the default CMR JSON returns. I worked a bit during the hackweek on making this easier via earthaccess (see https://github.com/nsidc/earthaccess/issues/305) @dshean could you provide a short code example for search -> geodataframe to ensure performance is the same if changing code in sliderule? 

 cc @jpswinski @JessicaS11 @tsutterley

**Comment by dshean** [comment]
Really appreciate you summarizing all of this @scottyhq!
Here is gist with sample code, observed runtimes, and region geojson: https://gist.github.com/dshean/b8f1c73dcf83938258363e722648da29
Let me know if anything else would be useful here.  Would be great to see some of this functionality upstream in earthaccess!
